### PR TITLE
Add calender component

### PR DESCRIPTION
**Pull request by Radian6405**
Pls test if it is possible to create any type of overlap in the events in the calender
 
 To test add this to main routes/+page.svelte (or any route) 
 ```
<script lang="ts">
	import Calender from '$lib/components/calender.svelte';

	let data = [];
</script>

<div class="p-2 bg-slate-950 ">
	<Calender events={data} />
</div>
```

for the `data` variable generate some event data from PockerBase dashboard and copy them as json
heres the data i used to test:
```
let data = [
		{
			banner: '',
			collectionId: 'pbc_1687431684',
			collectionName: 'events',
			created: '2024-12-28 13:09:11.634Z',
			description: '',
			endTime: '2024-12-31 13:45:00.000Z',
			id: 't1ky44h02436r27',
			location: 'OFG',
			sport: 'football',
			startTime: '2024-12-31 09:30:00.000Z',
			title: 'Semi-Finals',
			updated: '2024-12-28 13:09:11.634Z'
		},
		{
			banner: '',
			collectionId: 'pbc_1687431684',
			collectionName: 'events',
			created: '2024-12-28 13:25:28.443Z',
			description: '',
			endTime: '2024-12-31 14:00:00.000Z',
			id: '7dl5k89qmk37959',
			location: 'OFG',
			sport: 'football',
			startTime: '2024-12-31 12:00:00.000Z',
			title: 'Friendly match',
			updated: '2024-12-28 13:25:28.443Z'
		},
		{
			banner: '',
			collectionId: 'pbc_1687431684',
			collectionName: 'events',
			created: '2024-12-28 15:12:51.907Z',
			description: '',
			endTime: '2024-12-31 11:00:00.000Z',
			id: '68u690b59wj47c3',
			location: 'OFG',
			sport: 'football',
			startTime: '2024-12-31 09:00:00.000Z',
			title: 'idk',
			updated: '2024-12-28 15:12:51.907Z'
		},
		{
			banner: '',
			collectionId: 'pbc_1687431684',
			collectionName: 'events',
			created: '2024-12-28 15:13:49.238Z',
			description: '',
			endTime: '2024-12-28 12:00:00.000Z',
			id: '150f215mf64khe7',
			location: 'OFG',
			sport: 'football',
			startTime: '2024-12-11 11:00:00.000Z',
			title: 'idek',
			updated: '2024-12-28 15:13:49.238Z'
		},
		{
			banner: '',
			collectionId: 'pbc_1687431684',
			collectionName: 'events',
			created: '2024-12-28 15:23:11.855Z',
			description: '',
			endTime: '2024-12-28 17:00:00.000Z',
			id: 'np4czvftpwu8m1m',
			location: 'OFG',
			sport: 'football',
			startTime: '2024-12-28 13:00:00.000Z',
			title: 'man',
			updated: '2024-12-28 15:23:11.855Z'
		},
		{
			banner: '',
			collectionId: 'pbc_1687431684',
			collectionName: 'events',
			created: '2024-12-28 15:24:54.373Z',
			description: '',
			endTime: '2024-12-28 15:00:00.000Z',
			id: 'bg62gm68hytfr6j',
			location: 'OFG',
			sport: 'football',
			startTime: '2024-12-28 12:00:00.000Z',
			title: 'dude',
			updated: '2024-12-28 15:24:54.373Z'
		},
		{
			banner: '',
			collectionId: 'pbc_1687431684',
			collectionName: 'events',
			created: '2024-12-28 15:24:54.373Z',
			description: '',
			endTime: '2024-12-28 15:00:00.000Z',
			id: 'bg62gm68hytfr6j',
			location: 'OFG',
			sport: 'football',
			startTime: '2024-12-28 12:00:00.000Z',
			title: 'dude',
			updated: '2024-12-28 15:24:54.373Z'
		},
		{
			banner: '',
			collectionId: 'pbc_1687431684',
			collectionName: 'events',
			created: '2024-12-28 15:24:54.373Z',
			description: '',
			endTime: '2024-12-28 15:00:00.000Z',
			id: 'bg62gm68hytfr6j',
			location: 'OFG',
			sport: 'football',
			startTime: '2024-12-28 12:00:00.000Z',
			title: 'dude',
			updated: '2024-12-28 15:24:54.373Z'
		}
	];
```

### CODE REVIEW

**Review by skoriop — CHANGES_REQUESTED** [review]
Pretty nice work overall, it's responsive and seems to work fine with different numbers of columns and input sizes. There are a few minor styling nitpicks but I'll commit and push those myself.

The assignment of events to the grid could be simplified quite a bit though (which is what most of the requested changes are). Apart from that:
- [ ] It would make more sense to rename the files to `Calender.svelte` and `CalenderItem.svelte`, especially since that's how they are imported
- [ ] You should also change interface names to `UpperCamelCase`

**Review by soemone — COMMENTED** [review]
I think this is pretty good (Not that I understand how this UI works to the fullest lol)
But a few things I noticed:
- There's a different amount of space between the vertical and horizontal parts of the event component as you can see below (maybe padding or margins contribute to it? - I notice line 82 of calender.svelte `pr-2` provides proper padding to the top, matching the horizontal padding, this breaks the vertical padding). I don't know if this issue is isolated to my testing only though.
image
- Although this looks pretty good already, I think something like the UI found in teams would look even better, image for reference
image

File: app/src/lib/components/calender.svelte [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+<div class="flex flex-row bg-transparent p-2 text-white">
+	<!-- Time stamps -->
+	<div class="flex flex-col text-xs">
+		{#each { length: rows }, i}
+			<div class="h-20 w-10 pr-1 text-right">
+				{((i + calenderStartHour - 1) % 12) + 1}
+				{i < 12 - calenderStartHour ? 'AM' : 'PM'}
+			</div>
+		{/each}
+		<div class="w-10 pr-1 text-right">
```
Comment (soemone) [inline app/src/lib/components/calender.svelte:82]: Here is a possible offender to the differing padding / margins applied on the event view

Comment (skoriop) [reply app/src/lib/components/calender.svelte]: The actual cause was this line:

https://github.com/crux-bphc/arena-app/blob/bd7bdc1d3323f55e68ef89b7e1cf22a06629e938/app/src/lib/components/CalenderItem.svelte#L15

It's fixed in 2e9b891.

**Review by skoriop — APPROVED** [review]
LGTM

**Pull request merged**